src/plugins/data/public/actions/filters/create_filters_from_range_select.ts

```diff
@@ -31,11 +31,12 @@ export async function createFiltersFromRangeSelectAction(event: RangeSelectConte
     return [];
   }
 
-  const indexPattern = await getIndexPatterns().get(column.meta.indexPatternId);
-  const aggConfig = deserializeAggConfig({
-    ...column.meta,
-    indexPattern,
-  });
+  const { indexPatternId, ...aggConfigs } = column.meta.sourceParams;
+  const indexPattern = await getIndexPatterns().get(indexPatternId);
+  const aggConfigsInstance = getSearchService().aggs.createAggConfigs(indexPattern, [
+    aggConfigs as AggConfigSerialized,
```

Comment:
Maybe down the road we could look into introducing more specific typings for different sources. So that someone could do something like:

`column.meta.sourceParams as EsaggsDatatableSourceParams`

I guess this doesn't matter a ton though with the plan of moving toward utility functions, as most folks won't need to worry about these types at all.